### PR TITLE
Support mount propagation validation

### PR DESCRIPTION
**Pull request by miaoyq**
Part of #131.

/cc @feiskyer @Random-Liu 

Signed-off-by: Yanqiang Miao <miao.yanqiang@zte.com.cn>

### CODE REVIEW

**Comment by Random-Liu** [comment]
@miaoyq Does cri-containerd pass this?

**Comment by miaoyq** [comment]
@Random-Liu Yes, have passed.

**Comment by Random-Liu** [comment]
@miaoyq Cool! Will review it now.

File: pkg/validate/container.go [hunk 1]
```diff
@@ -201,6 +203,117 @@ var _ = framework.KubeDescribe("Container", func() {
...
+		})
+
+		AfterEach(func() {
+			By("stop PodSandbox")
+			rc.StopPodSandbox(podID)
+			By("delete PodSandbox")
+			rc.RemovePodSandbox(podID)
+		})
+
+		It("mount with 'rprivate' should not support propagation", func() {
```
Comment (Random-Liu) [inline pkg/validate/container.go]: We should also check mount inside container could not be propagated to host.

File: pkg/validate/container.go [hunk 2]
```diff
@@ -201,6 +203,117 @@ var _ = framework.KubeDescribe("Container", func() {
...
+			command = []string{"sh", "-c", "mount --bind /etc " + containerMntPoint}
+			execSyncContainer(rc, containerID, command)
+
+			By("check whether containerMntPoint contains file or dir in host")
+			fileInfo, err := ioutil.ReadDir(containerMntPoint)
+			framework.ExpectNoError(err, "failed to ReadDir %q in Host", containerMntPoint)
+			Expect(len(fileInfo)).NotTo(BeZero(), "len(fileInfo) should not be zero.")
+		})
+
+		It("mount with 'rslave' should support propagation from host to container", func() {
```
Comment (Random-Liu) [inline pkg/validate/container.go]: We should also check mount inside container could not be propagated to host.

File: pkg/validate/container.go [hunk 3]
```diff
@@ -201,6 +203,117 @@ var _ = framework.KubeDescribe("Container", func() {
...
+			By("create a propatation mount point in host")
+			createPropagationMountPoint(propagationSrcDir, propagationMntPoint)
+
+			By("check whether propagationMntPoint contains file or dir in container")
+			command := []string{"ls", "-A", propagationMntPoint}
+			output := execSyncContainer(rc, containerID, command)
+			Expect(len(output)).NotTo(BeZero(), "len(output) should not be zero.")
+		})
+
+		It("mount without propagation option should not support propagation", func() {
```
Comment (Random-Liu) [inline pkg/validate/container.go]: This is not defined in CRI, runtime may return an error for non-supported mount propagation option, so let's not test this.

File: pkg/validate/container.go (hunk 1 above)
Comment (miaoyq) [reply pkg/validate/container.go]: Will do.

File: pkg/validate/container.go (hunk 2 above)
Comment (miaoyq) [reply pkg/validate/container.go]: Will do.

File: pkg/validate/container.go (hunk 3 above)
Comment (miaoyq) [reply pkg/validate/container.go]: OK, will remove this part.

**Comment by miaoyq** [comment]
@Random-Liu Updated, cri-containerd have passed also.

**Comment by Random-Liu** [comment]
LGTM

**Comment by feiskyer** [comment]
LGTM

**Pull request merged**